### PR TITLE
Add more unit tests

### PR DESCRIPTION
**Pull request by diversemix**

### CODE REVIEW

File: src/index.test.ts [hunk 1]
```diff
@@ -21,14 +21,26 @@ describe('cli', () => {
...
             commands,
         );
     });
 
-    it('command init is called in the cli constructor', () => {
-        expect(commands.init).toHaveBeenCalledTimes(1);
+    it('generates unique filenames over 1 second', done => {
+        const thing1: string = cli.filePath('thing1');
+        let thing2: string;
+        setTimeout(() => {
```
Comment (diversemix) [inline src/index.test.ts:36]: This looks a bit bad - but the spec of the filePath command is to only be guaranteed unique >1 second

File: src/cli.ts [hunk 2]
```diff
@@ -32,7 +34,11 @@ export class Cli {
             // eslint-disable-next-line @typescript-eslint/no-explicit-any
         } as any;
```
Comment (Cooryd) [inline src/cli.ts]: there's a `@types/umzug` package I think we should be using instead of linting suppression

File: src/commands.ts [hunk 3]
```diff
@@ -1,34 +1,34 @@
 import * as table from 'borderless-table';
 import { writeFileSync } from 'fs';
 import { EOL } from 'os';
-import { Migration, Umzug } from 'umzug';
 import migrationTemplate from './migration-template';
+import { MigrationTool, Migration, OutputStream } from 'types';
```
Comment (Cooryd) [inline src/commands.ts]: should this be `./types`?

File: src/mock-tool.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        this.commands.push('pending');
+        return Promise.resolve([]);
+    }
+
+    executed(): Promise<Migration[]> {
+        this.commands.push('executed');
+        return Promise.resolve([]);
+    }
+
+    up(migration?: string | undefined): Promise<Migration[]> {
```
Comment (Cooryd) [inline src/mock-tool.ts]: does this still need type undefined if if it's optional?

File: src/commands.ts [hunk 5]
```diff
@@ -1,34 +1,34 @@
...
 
     public async rollback(): Promise<void> {
-        const migrations = await this.umzug.down();
+        const migrations = await this.tool.down();
 
-        this.printMigrations(migrations, 'Successfuly rolled back:');
+        this.printMigrations(migrations, 'Successfully rolled back:');
     }
 
     public async showStatus({ pending, executed }): Promise<void> {
```
Comment (erezmus) [inline src/commands.ts]: suggestion: should probably add boolean types

File: src/commands.ts [hunk 6]
```diff
@@ -37,22 +37,22 @@ export class Commands {
...
         }
 
         if (executed) {
-            const executedMigrations = await this.umzug.executed();
+            const executedMigrations = await this.tool.executed();
             this.printMigrations(executedMigrations, 'Executed migrations');
         }
     }
 
     private printLn(msg = ''): void {
```
Comment (erezmus) [inline src/commands.ts:44]: suggestion: could also add string type

File: src/commands.ts (hunk 3 above)
Comment (diversemix) [reply src/commands.ts]: it could be - is this your preferred style?

**Review by erezmus — APPROVED** [review]

File: src/commands.ts (hunk 5 above)
Comment (diversemix) [reply src/commands.ts]: Where?

Comment (erezmus) [reply src/commands.ts]: sorry, for the `pending` and `executed` parameters in showStatus.

File: src/commands.ts (hunk 6 above)
Comment (erezmus) [reply src/commands.ts:44]: in printLn

File: src/cli.ts (hunk 2 above)
Comment (diversemix) [reply src/cli.ts]: you happy with this now?

File: src/cli.ts [hunk 7]
```diff
@@ -19,20 +19,28 @@ type statusCommandArguments = yargs.Arguments<
...
             storage: 'knex-umzug',
             storageOptions: {
                 context: 'default',
                 connection,
                 tableName: 'migrations',
             },
             migrations: { ...this.options.migrations, params: [connection] },
+            // This cannot be of type UmzugOptions as its from `knex-umzug`
+            // which doesn't have the correct type.
             // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (Cooryd) [inline src/cli.ts]: can this b removed now then?

**Pull request merged**